src/internal/m365/collection/site/handlers.go

```diff
@@ -20,3 +20,32 @@ type getItemByIDer interface {
 type getItemser interface {
 	GetItems(ctx context.Context, cc api.CallConfig) ([]models.Listable, error)
 }
+
+type restoreHandler interface {
+	PostLister
+	PostListItemer
+	DeleteLister
+}
+
+type PostLister interface {
+	PostList(
+		ctx context.Context,
+		listName string,
+		oldListByteArray []byte,
+	) (models.Listable, error)
+}
+
+type PostListItemer interface {
+	PostListItem(
+		ctx context.Context,
+		listID string,
+		oldListByteArray []byte,
+	) ([]models.ListItemable, error)
```

Comment:
hmm, I have some api conerns here.  The name suggests you're posting one item.  The return value has many items.  That doesn't match up.

It looks like (according to the name) that you're feeding this func the same byte slice as you give the PostList func.  That means that one of the following is true:
1. all items are contained in that slice, and their creation should be handled within `PostList` instead of running a two-step update function.
2. this is a different slice of bytes, and should be named differently to match.

I assume 1/ is true.  I would suggest packing the full restore process into that function instead of splitting it out.  That way this layer is only concerned with the complete creation of a list using the original byte slice, and the api can deal with the details.

Reply:
Yes the first point is correct. 
Sure let me put the PostItem into PostList. 
I will change the function schema as well:
- Name of the function to `PostListItems`
- Accepts `models.Listable` and not `listID` & `oldListByteArray`